x-pack/plugins/alerting/server/rules_client/lib/untrack_rule_alerts.ts

```diff
@@ -15,30 +15,40 @@ import { EVENT_LOG_ACTIONS } from '../../plugin';
 import { createAlertEventLogRecordObject } from '../../lib/create_alert_event_log_record_object';
 import { RulesClientContext } from '../types';
 
-export const recoverRuleAlerts = async (
+export const untrackRuleAlerts = async (
   context: RulesClientContext,
   id: string,
   attributes: RawRule
 ) => {
-  return withSpan({ name: 'recoverRuleAlerts', type: 'rules' }, async () => {
+  return withSpan({ name: 'untrackRuleAlerts', type: 'rules' }, async () => {
     if (!context.eventLogger || !attributes.scheduledTaskId) return;
     try {
-      const { state } = taskInstanceToAlertTaskInstance(
+      const taskInstance = taskInstanceToAlertTaskInstance(
         await context.taskManager.get(attributes.scheduledTaskId),
         attributes as unknown as SanitizedRule
       );
 
-      const recoveredAlerts = mapValues<Record<string, RawAlert>, Alert>(
+      const { state } = taskInstance;
+
+      const untrackedAlerts = mapValues<Record<string, RawAlert>, Alert>(
         state.alertInstances ?? {},
         (rawAlertInstance, alertId) => new Alert(alertId, rawAlertInstance)
       );
-      const recoveredAlertIds = Object.keys(recoveredAlerts);
 
-      for (const alertId of recoveredAlertIds) {
-        const { group: actionGroup } = recoveredAlerts[alertId].getLastScheduledActions() ?? {};
-        const instanceState = recoveredAlerts[alertId].getState();
-        const message = `instance '${alertId}' has recovered due to the rule was disabled`;
-        const alertUuid = recoveredAlerts[alertId].getUuid();
+      const untrackedAlertIds = Object.keys(untrackedAlerts);
+
+      const { isLifecycleAlert } = taskInstance.state?.alertTypeState ?? {
```

Comment:
When a rule type registers with the rule type registery, it specifies an `autoRecoverAlerts` which defaults to `true`. We should use this to determine whether this is a lifecycle alert instead of relying on the rule types to return this flag from its executors. `autoRecoverAlerts = true` indicates a lifecycle rule type.